js/src/dom/data.js

```diff
@@ -11,57 +11,16 @@
  * ------------------------------------------------------------------------
  */
 
-const mapData = (() => {
-  const storeData = {}
-  let id = 1
-  return {
-    set(element, key, data) {
-      if (typeof element.bsKey === 'undefined') {
-        element.bsKey = {
-          key,
-          id
-        }
-        id++
-      }
+const mapData = new Map()
 
-      storeData[element.bsKey.id] = data
-    },
-    get(element, key) {
-      if (!element || typeof element.bsKey === 'undefined') {
-        return null
-      }
-
-      const keyProperties = element.bsKey
-      if (keyProperties.key === key) {
-        return storeData[keyProperties.id]
-      }
-
-      return null
-    },
-    delete(element, key) {
-      if (typeof element.bsKey === 'undefined') {
-        return
-      }
-
-      const keyProperties = element.bsKey
-      if (keyProperties.key === key) {
-        delete storeData[keyProperties.id]
-        delete element.bsKey
-      }
-    }
-  }
-})()
-
-const Data = {
-  setData(instance, key, data) {
-    mapData.set(instance, key, data)
+export default {
+  setData(element, data) {
```

Comment:
Maybe we should rename the objects to `set`, `get` and `remove` @Johann-S? I mean, we use them as `Data.foo` so having `Data/fooData` seems a little redundant. Just thinking out loud here :)

Reply:
yep I agree too 👍